comptoir-ejb/src/main/java/be/valuya/comptoir/service/SaleService.java

```diff
@@ -272,16 +325,56 @@ public Sale saveSale(Sale sale) {
             sale.setVatAmount(BigDecimal.ZERO);
         }
 
-        BigDecimal vatExclusiveAmout = sale.getVatExclusiveAmout();
+        BigDecimal vatExclusiveAmout = sale.getVatExclusiveAmount();
         if (vatExclusiveAmout == null) {
-            sale.setVatExclusiveAmout(BigDecimal.ZERO);
+            sale.setVatExclusiveAmount(BigDecimal.ZERO);
         }
 
         return entityManager.merge(sale);
     }
 
+    public void cancelOpenSale(Sale sale) {
+        if (sale.isClosed()) {
+            throw new IllegalArgumentException("The sale is closed");
+        }
+        List<ItemSale> itemSaleList = findSaleItems(sale);
+        for (ItemSale itemSale : itemSaleList) {
+            removeItemSale(itemSale);
+        }
+        Sale managedSale = entityManager.merge(sale);
+        entityManager.remove(managedSale);
```

Comment:
I though that for open sales we could remove them completely from the database. I might be wrong, I didn't check all linked entitites were correctly removed, especially the accounting-related ones created in the calc* method.